the macros dont detect neon support based on target, so we have to prevent its use on armhf
diff --git a/lib/arm/cpu_features.h b/lib/arm/cpu_features.h
index 548d31e..c6e52ba 100644
--- a/lib/arm/cpu_features.h
+++ b/lib/arm/cpu_features.h
@@ -137,11 +137,12 @@ static inline u32 get_arm_cpu_features(void) { return 0; }
  */
 #if HAVE_CRC32_NATIVE || \
 	(HAVE_DYNAMIC_ARM_CPU_FEATURES && \
-	 (__has_builtin(__builtin_arm_crc32b) || \
+	 (__ARM_ARCH >= 7 && \
+	 (__has_builtin(__DEFINE_UNALIGNED_TYPEbuiltin_arm_crc32b) || \
 	  GCC_PREREQ(11, 3) || \
 	  (GCC_PREREQ(10, 4) && !GCC_PREREQ(11, 0)) || \
 	  (GCC_PREREQ(9, 5) && !GCC_PREREQ(10, 0)) || \
-	  defined(_MSC_VER)))
+	  defined(_MSC_VER))))
 #  define HAVE_CRC32_INTRIN	1
 #else
 #  define HAVE_CRC32_INTRIN	0
